diff --git a/src/pdm/backend/_vendor/pyproject_metadata/__init__.py b/src/pdm/backend/_vendor/pyproject_metadata/__init__.py
index df826f8..7f69f3a 100644
--- a/src/pdm/backend/_vendor/pyproject_metadata/__init__.py
+++ b/src/pdm/backend/_vendor/pyproject_metadata/__init__.py
@@ -24,11 +24,11 @@ if typing.TYPE_CHECKING:
     else:
         from typing import Self
 
-import packaging.markers
-import packaging.requirements
-import packaging.specifiers
-import packaging.utils
-import packaging.version
+import pdm.backend._vendor.packaging.markers as pkg_markers
+import pdm.backend._vendor.packaging.requirements as pkg_requirements
+import pdm.backend._vendor.packaging.specifiers as pkg_specifiers
+import pdm.backend._vendor.packaging.utils as pkg_utils
+import pdm.backend._vendor.packaging.version as pkg_version
 
 
 __version__ = '0.8.0'
@@ -175,11 +175,11 @@ class Readme(typing.NamedTuple):
 @dataclasses.dataclass
 class StandardMetadata:
     name: str
-    version: packaging.version.Version | None = None
+    version: pkg_version.Version | None = None
     description: str | None = None
     license: License | None = None
     readme: Readme | None = None
-    requires_python: packaging.specifiers.SpecifierSet | None = None
+    requires_python: pkg_specifiers.SpecifierSet | None = None
     dependencies: list[Requirement] = dataclasses.field(default_factory=list)
     optional_dependencies: dict[str, list[Requirement]] = dataclasses.field(default_factory=dict)
     entrypoints: dict[str, dict[str, str]] = dataclasses.field(default_factory=dict)
@@ -202,7 +202,7 @@ class StandardMetadata:
 
     @property
     def canonical_name(self) -> str:
-        return packaging.utils.canonicalize_name(self.name)
+        return pkg_utils.canonicalize_name(self.name)
 
     @classmethod
     def from_pyproject(
@@ -235,7 +235,7 @@ class StandardMetadata:
 
         version_string = fetcher.get_str('project.version')
         requires_python_string = fetcher.get_str('project.requires-python')
-        version = packaging.version.Version(version_string) if version_string else None
+        version = pkg_version.Version(version_string) if version_string else None
 
         if version is None and 'version' not in dynamic:
             msg = 'Field "project.version" missing and "version" not specified in "project.dynamic"'
@@ -256,7 +256,7 @@ class StandardMetadata:
             description,
             cls._get_license(fetcher, project_dir),
             cls._get_readme(fetcher, project_dir),
-            packaging.specifiers.SpecifierSet(requires_python_string) if requires_python_string else None,
+            pkg_specifiers.SpecifierSet(requires_python_string) if requires_python_string else None,
             cls._get_dependencies(fetcher),
             cls._get_optional_dependencies(fetcher),
             cls._get_entrypoints(fetcher),
@@ -358,15 +358,15 @@ class StandardMetadata:
         requirement = copy.copy(requirement)
         if requirement.marker:
             if 'or' in requirement.marker._markers:
-                requirement.marker = packaging.markers.Marker(
+                requirement.marker = pkg_markers.Marker(
                     f'({requirement.marker}) and extra == "{extra}"'
                 )
             else:
-                requirement.marker = packaging.markers.Marker(
+                requirement.marker = pkg_markers.Marker(
                     f'{requirement.marker} and extra == "{extra}"'
                 )
         else:
-            requirement.marker = packaging.markers.Marker(f'extra == "{extra}"')
+            requirement.marker = pkg_markers.Marker(f'extra == "{extra}"')
         return requirement
 
     @staticmethod
@@ -462,8 +462,8 @@ class StandardMetadata:
         requirements: list[Requirement] = []
         for req in requirement_strings:
             try:
-                requirements.append(packaging.requirements.Requirement(req))
-            except packaging.requirements.InvalidRequirement as e:
+                requirements.append(pkg_requirements.Requirement(req))
+            except pkg_requirements.InvalidRequirement as e:
                 msg = (
                     'Field "project.dependencies" contains an invalid PEP 508 '
                     f'requirement string "{req}" ("{e}")'
@@ -502,8 +502,8 @@ class StandardMetadata:
                     )
                     raise ConfigurationError(msg)
                 try:
-                    requirements_dict[extra].append(packaging.requirements.Requirement(req))
-                except packaging.requirements.InvalidRequirement as e:
+                    requirements_dict[extra].append(pkg_requirements.Requirement(req))
+                except pkg_requirements.InvalidRequirement as e:
                     msg = (
                         f'Field "project.optional-dependencies.{extra}" contains '
                         f'an invalid PEP 508 requirement string "{req}" ("{e}")'
