diff --git a/src/pdm/backend/_vendor/pyproject_metadata/__init__.py b/src/pdm/backend/_vendor/pyproject_metadata/__init__.py
index 39ae6e4..283264b 100644
--- a/src/pdm/backend/_vendor/pyproject_metadata/__init__.py
+++ b/src/pdm/backend/_vendor/pyproject_metadata/__init__.py
@@ -63,6 +63,7 @@ if typing.TYPE_CHECKING:

     from .project_table import Dynamic, PyProjectTable

+import pdm.backend._vendor.packaging as packaging
 import packaging.markers
 import packaging.specifiers
 import packaging.utils
diff --git a/src/pdm/backend/_vendor/pyproject_metadata/pyproject.py b/src/pdm/backend/_vendor/pyproject_metadata/pyproject.py
index d1822e1..a85f9a1 100644
--- a/src/pdm/backend/_vendor/pyproject_metadata/pyproject.py
+++ b/src/pdm/backend/_vendor/pyproject_metadata/pyproject.py
@@ -13,6 +13,7 @@ import pathlib
 import re
 import typing

+import pdm.backend._vendor.packaging as packaging
 import packaging.requirements

 from .errors import ErrorCollector
